### DISCUSSION TITLE
Help, IISNODE problem

### DISCUSSION THREAD

**Discussion by AlexVanK** [directus/directus#13159] [Archive]
Hello,

I tried to update Directus from 9.8.0 to 9.10.0
and somehow I broke iisnode

Steps:

1. Opened CMD, and run npm update
2. Got an error, something to do with permissions
3. Opened CMD as admin, and run npm update
4. Got a similar error on another module
5. Wiped node_modules folder
6. Run npm update again (not npm install)
Went smooth.

Server starts:

![image](https://user-images.githubusercontent.com/56647386/167224497-f780702b-0a75-43e3-9dfe-be14e1b7d0e2.png)

However upon opening the app through the IIS binded IP
![image](https://user-images.githubusercontent.com/56647386/167224547-538c4978-b57e-42ed-84d7-1d375e4c5e88.png)

**Comment by rijkvanzanten** [comment]
What was the full error thrown? The line shown in the screenshot seems to be just a warning about something else

**Comment by reggiedunloper** [comment]
You may be running into the same issue I faced as mentioned in #12998.

**Reply by AlexVanK** [reply]
This helped me pinpoint the problem, thank you.
I will respond under your issue.

**Comment by mickert** [comment]
I recently ranninto this and other Directis startup issues on IIS with IISNode and fixed / worked-around them.

I created the following gist with the working startup app.js / index.js and a fix for startServer(). I'll try to make a PR for this simple startServer fix when i find some time and energy.


[Directus 9.9, 10 and up running from app.js or index.js on IIS, IISNODE, PLESK, etc. (fix for Package subpath './server' is not defined by "exports", DeprecationWarning: Buffer()) and ERR_SOCKET_BAD_PORT](https://gist.github.com/mickert/786a03707de3eaaee3b34a350409fb6d) 
https://gist.github.com/mickert/786a03707de3eaaee3b34a350409fb6d

Sorry to revive this old thread, but hopefully it benefits someone.
